### PR TITLE
Fixing PMD rules. Reducing complexity and simplifying code.

### PR DESCRIPTION
**Pull request by Delawen**
PMD was complaining. I did a couple refactorings more to avoid unused variable assignments. deprecated functions and such.

### CODE REVIEW

File: app/integration/runtime/src/main/java/io/syndesis/integration/runtime/IntegrationRouteBuilder.java [hunk 1]
```diff
@@ -287,23 +287,18 @@ private void configureFlow(Flow flow, final String flowIndex) throws URISyntaxEx
...
         } else {
             // not in a pipeline so set the step id header to be sure it is there
-            definition = parent.setHeader(IntegrationLoggingConstants.STEP_ID, constant(stepId));
+            parent.setHeader(IntegrationLoggingConstants.STEP_ID, constant(stepId));
         }
 
-        definition = parent.process(OutMessageCaptureProcessor.INSTANCE)
+        return parent.process(OutMessageCaptureProcessor.INSTANCE)
                         .id(String.format("capture-out:%s", stepId));
-        return definition;
```
Comment (Delawen) [inline app/integration/runtime/src/main/java/io/syndesis/integration/runtime/IntegrationRouteBuilder.java]: definition variable was not really used. We can avoid assigning and preventing weird behaviours in the future.

File: app/integration/runtime/src/main/java/io/syndesis/integration/runtime/IntegrationRouteBuilder.java [hunk 2]
```diff
@@ -330,7 +325,7 @@ private void configureFlow(Flow flow, final String flowIndex) throws URISyntaxEx
                     }
                 }
             }
-            if (onException.isPresent() && onExceptionStep != null) {
```
Comment (Delawen) [inline app/integration/runtime/src/main/java/io/syndesis/integration/runtime/IntegrationRouteBuilder.java]: Kind of a duplicated useless check. If onExceptionStep is not null, then onException is always present.

File: app/integration/runtime/src/main/java/io/syndesis/integration/runtime/IntegrationRouteBuilder.java [hunk 3]
```diff
@@ -287,23 +287,18 @@ private void configureFlow(Flow flow, final String flowIndex) throws URISyntaxEx
             }
         } else {
             // not in a pipeline so set the step id header to be sure it is there
-            definition = parent.setHeader(IntegrationLoggingConstants.STEP_ID, constant(stepId));
+            parent.setHeader(IntegrationLoggingConstants.STEP_ID, constant(stepId));
```
Comment (lburgazzoli) [inline app/integration/runtime/src/main/java/io/syndesis/integration/runtime/IntegrationRouteBuilder.java]: `setHeader` returns a `ProcessorDefinition<?> `and it is true that in this case it returns the same instance but reassigning it to a definition is IMHO more idiomatic as the underlying code is the free to return a different instance.

File: app/integration/runtime/src/main/java/io/syndesis/integration/runtime/IntegrationRouteBuilder.java [hunk 4]
```diff
@@ -287,23 +287,18 @@ private void configureFlow(Flow flow, final String flowIndex) throws URISyntaxEx
...
-        definition = parent.process(OutMessageCaptureProcessor.INSTANCE)
+        return parent.process(OutMessageCaptureProcessor.INSTANCE)
                         .id(String.format("capture-out:%s", stepId));
-        return definition;
     }
 
     private ProcessorDefinition<?> configureRouteDefinition(ProcessorDefinition<?> definition, Flow flow, String flowId) {
-        if (definition instanceof RouteDefinition) {
+        if (definition instanceof RouteDefinition
+                && !containsConfiguredActivityTrackingPolicies((RouteDefinition)definition)) {
```
Comment (lburgazzoli) [inline app/integration/runtime/src/main/java/io/syndesis/integration/runtime/IntegrationRouteBuilder.java]: You may move the check `definition instanceof RouteDefinition` to the `containsConfiguredActivityTrackingPolicies` method

Comment (Delawen) [reply app/integration/runtime/src/main/java/io/syndesis/integration/runtime/IntegrationRouteBuilder.java]: True!

File: app/integration/runtime/src/main/java/io/syndesis/integration/runtime/IntegrationRouteBuilder.java (hunk 2 above)
Comment (lburgazzoli) [reply app/integration/runtime/src/main/java/io/syndesis/integration/runtime/IntegrationRouteBuilder.java]: You still have to check it as at line 321 `onException` is set to Optional.empty();

**Pull request merged**